src/Context.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Drupal\commerce;
+
+use Drupal\commerce_store\Entity\StoreInterface;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Session\AccountInterface;
+
+final class Context {
+
+  /**
+   * The user entity.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $user;
+
+  /**
+   * The store entity.
+   *
+   * @var \Drupal\commerce_store\Entity\Store
+   */
+  protected $store;
+
+  /**
+   * The date.
+   *
+   * @var \Drupal\Core\Datetime\DrupalDateTime
+   */
+  protected $date;
+
+  /**
+   * Constructs a new Commerce Context object.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $user
+   *   The user entity.
+   * @param \Drupal\commerce_store\Entity\StoreInterface $store
+   *   The store entity.
+   * @param \Drupal\Core\Datetime\DrupalDateTime $date
+   *   The date.
+   */
+  public function __construct(AccountInterface $user, StoreInterface $store, DrupalDateTime $date) {
+    $this->user = $user;
+    $this->store = $store;
+    $this->date = $date;
```

Comment:
Can we default to the current date here if $date is NULL? Would simplify callers.